### DISCUSSION TITLE
Get / Set the current design standard

### DISCUSSION THREAD

**Discussion by NathanMSY** [Dlubal-Software/Dlubal_CSharp_Client#51] [Q&A] [answered]
Is there any possibility to set the design standard of e.g. the steel-addon? (see below) It used to be possible with the RFEM5 API and was a crucial functionality but I cant find a function / API call to set the standard for the addons except for the load combination wizard.
We can't port our apps to the latest generation of Dlubal programs if features like this are missing. There is no workaround.

![2024-01-11 11_54_37-MSYConfigurator - Microsoft Visual Studio](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/75130759/3421d07e-0950-4fae-9858-2c0d25830ca8)

**Comment by Rebecca-Coding21** [comment, marked as answer]
Hi @NathanMSY,
unfortunately there is no simple function to set the annex/standard for add-ons at the moment. Nevertheless there is a workaround you can use:
```cs
string standard = "EN 1993 | NBN | 2019-08";

// activate steel design
var addOnStatus = new addon_list_type()
{
    design_addons = new addon_list_design_addons_list_type()
    {
        steel_design_active = true,
        steel_design_activeSpecified = true,
    }
};

try
{
    model.begin_modification("Geometry");
    //model.set_section(sectionBuildtUp);
    model.set_addon_statuses(addOnStatus);

    // change standard of steel design
    byte[] standardByte = new UTF8Encoding(true).GetBytes($"general.current_standard_for_steel_design = '{standard}'");
    using (FileStream file = File.Open($"{Directory.GetCurrentDirectory()}./standard.js", FileMode.Create, FileAccess.Write))
    {
        file.Write(standardByte);
    }

    model.run_script($"{CurrentDirectory}./standard.js");
}
...
```
We will consider adding such function for future development.

**Comment by NathanMSY** [comment]
Thanks for the reply, Rebecca.
That's a nice workaround. I assume that this could be a workaround for any function that is missing in the C# Library but is available via scripting.
Really helpful!
